terraform/root_integration.tf

```diff
@@ -0,0 +1,102 @@
+  project               = "tdr"
+  alb_dns_name          = module.jenkins_alb.alb_dns_name
+  alb_zone_id           = module.jenkins_alb.alb_zone_id
+  a_record_name         = "jenkins"
+}
+
+module "jenkins_ec2" {
+  source              = "./tdr-terraform-modules/ec2"
+  ami_id              = module.jenkins_ami.encrypted_ami_id
+  common_tags         = local.common_tags
+  environment         = local.environment
+  name                = local.ec2_instance_name
+  subnet_id           = module.jenkins_vpc.private_subnets[1]
+  security_group_id   = module.jenkins_ec2_security_group.security_group_id
+  attach_policies     = { ec2_policy = module.jenkins_ec2_policy.policy_arn }
+  private_ip          = "10.0.1.221"
+  user_data           = "user_data_jenkins_docker"
+  user_data_variables = { jenkins_cluster_name = "jenkins-${local.environment}" }
+  instance_type       = "t2.medium"
+  volume_size         = 60
+}
+
+# Configure Jenkins backup using Systems Manager Maintenance Windows
+module "jenkins_backup_maintenance_window" {
+  source          = "./tdr-terraform-modules/ssm_maintenance_window"
+  command         = "docker exec $(docker ps -aq -f ancestor=${module.ecr_jenkins_repository.repository.repository_url} -f status=running) /opt/backup.sh ${data.aws_ssm_parameter.jenkins_backup_healthcheck_url.value}"
+  ec2_instance_id = module.jenkins_ec2.instance_id
+  name            = "tdr-jenkins-backup-window"
+  schedule        = "cron(0 0 18 ? * MON-FRI *)"
+  common_tags     = local.common_tags
+}
+
+module "jenkins_maintenance_window_event" {
+  source                  = "./tdr-terraform-modules/cloudwatch_events"
+  event_pattern           = "jenkins_maintenance_event_window"
+  lambda_event_target_arn = list(data.aws_lambda_function.notifications_function.arn)
+  rule_name               = "jenkins-backup-maintenance-window"
+  rule_description        = "Capture failed runs of the jenkins backup"
+  event_variables         = { window_id = module.jenkins_backup_maintenance_window.window_id }
+}
+
+module "jenkins_integration_fargate_policy" {
+  source        = "./tdr-terraform-modules/iam_policy"
+  name          = "TDRJenkinsFargatePolicy${title(local.environment)}"
+  policy_string = templatefile("./tdr-terraform-modules/iam_policy/templates/jenkins_fargate_integration.json.tpl", { account_id = data.aws_caller_identity.current.account_id })
+}
+
+module "jenkins_integration_fargate_role" {
+  source             = "./tdr-terraform-modules/iam_role"
+  common_tags        = local.common_tags
+  assume_role_policy = templatefile("./tdr-terraform-modules/iam_policy/templates/assume_role_policy.json.tpl", { role_arn = module.jenkins_integration_ecs_task_role.role.arn })
+  name               = "TDRJenkinsFargateRole${title(local.environment)}"
+  policy_attachments = { fargate_policy = module.jenkins_integration_fargate_policy.policy_arn, ssm_core = "arn:aws:iam::aws:policy/AmazonSSMManagedInstanceCore" }
+}
+
+module "jenkins_certificate" {
```

Comment:
Not sure how the certificates work, but do we run the risk of having a name clash for the prod/staging instance, if they also need a certificate?

Reply:
I'm going to call the prod one `jenkins-prod.tdr-management...` (I will consider other names)